AEFeinstein/Swadge-IDF-5.0
# Record/Playback Emulator Inputs

### Description

This adds functionality to the emulator for recording inputs to a file and playing them back later. It also supports some extra non-input actions when playing back a recording, such as enabling fuzzing, saving a screenshot, force-changing modes, and exiting the emulator.

Oh, and it adds a headless argument to have a hidden window, but I'm not sure if that works outside of Linux / for all rawdraw drivers.

### Test Instructions

Record inputs by running `swadge_emulator --record recording.csv` or just `swadge_emulator --record` (which should name the file `rec-xxxxxxx.csv`). When you're done, close the emulator and the file should now be full of inputs!

Play back that file with `swadge_emulator --playback recording.csv` and it should do exactly the same things as when recorded.

Edit the recording, and try adding these lines to the end:
```
14500000,SetMode,Touch Test
15000000,Screenshot,
15000000,Fuzz,
17500000,Screenshot,filename.bmp
20000000,Quit,
```
If you play back the recording with `swadge_emulator --playback recording.csv` again, it should repeat your inputs and then, after the original script ends, will wait until almost 15 seconds have passed and switch to the touch test mode, take a screenshot (`screenshot-xx.bmp`), begin fuzzing, then take another screenshot 2.5s later, and finally quit after waiting another 2.5s. You should see a `screenshot-xx.bmp` and `filename.bmp` in your working directory, and they should look correct.

Then, delete the screenshots and run the emulator again with `--headless` as well; it should do everything the same, but no window should appear. The screenshots should still be generated and look normal.

### Ticket Links



### Readiness Checklist
